### PR TITLE
Transpose sinking general

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
- unite all transformations into general
- fix TransposeFuse transformation setting up Transpose Constant element type
- add code that prevents moving backward Transposes

### Tickets:
 - 92896

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp [hunk 1]
```diff
@@ -293,15 +294,21 @@ ov::pass::TransposeFuse::TransposeFuse() {
                 is_ordered = false;
         }
 
+        auto transpose_oder_type = transpose1_order->get_element_type();
```
Comment (nkogteva) [inline src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp:297]:
```suggestion
        auto transpose_order_type = transpose1_order->get_element_type();
```

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_utils.cpp [hunk 2]
```diff
@@ -169,4 +169,59 @@ NodeVector InsertTransposeBeforeNode(NodePtr main_node, std::shared_ptr<Constant
...
+        for (auto& consumer_input : node->output(i).get_target_inputs()) {
+            if (!CanPropagateForwardThrough(consumer_input.get_node()))
+                return false;
+        }
+    }
+
+    return true;
+}
+
+#define NO_TRANSPOSE_SINKING_KEY "no_transpose_sinking"
```
Comment (nkogteva) [inline src/common/transformations/src/transformations/common_optimizations/transpose_sinking_utils.cpp:205]: Does it make sense to create attribute based on `ov::RuntimeAttribute` ?

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_general.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ov::pass::TransposeSinkingGeneralBackward::TransposeSinkingGeneralBackward() {
+    MATCHER_SCOPE(TransposeSinkingGeneralBackward);
+    add_matcher<ov::pass::TransposeSinkingUnaryBackward>();
+    add_matcher<ov::pass::TransposeSinkingBinaryElementwiseBackward>();
+    add_matcher<ov::pass::TransposeSinkingConcatBackward>();
+    add_matcher<ov::pass::TransposeSinkingSplitBackward>();
+    add_matcher<ngraph::pass::TransposeFuse>();
+}
+
+bool ov::pass::TransposeSinkingGeneral::run_on_model(const std::shared_ptr<ov::Model>& f) {
```
Comment (nkogteva) [inline src/common/transformations/src/transformations/common_optimizations/transpose_sinking_general.cpp:39]:
```suggestion
bool ov::pass::TransposeSinkingGeneral::run_on_model(const std::shared_ptr<ov::Model>& f) {
    RUN_ON_FUNCTION_SCOPE(TransposeSinkingGeneral);
```

File: src/common/transformations/include/transformations/common_optimizations/transpose_sinking_utils.hpp [hunk 4]
```diff
@@ -47,4 +47,8 @@ ov::NodeVector InsertTransposeBeforeNode(std::shared_ptr<ov::Node> main_node,
                                          std::shared_ptr<ov::opset9::Constant> transpose_const);
 }  // namespace sink_backward
 
+bool IsSinkingEnable(std::shared_ptr<ov::Node>);
```
Comment (nkogteva) [inline src/common/transformations/include/transformations/common_optimizations/transpose_sinking_utils.hpp:50]:
```suggestion
bool IsSinkingEnabled(std::shared_ptr<ov::Node>);
```

File: src/common/transformations/include/transformations/common_optimizations/transpose_sinking_utils.hpp [hunk 5]
```diff
@@ -47,4 +47,8 @@ ov::NodeVector InsertTransposeBeforeNode(std::shared_ptr<ov::Node> main_node,
...
+bool IsSinkingEnable(ov::Node*);
```
Comment (nkogteva) [inline src/common/transformations/include/transformations/common_optimizations/transpose_sinking_utils.hpp:51]:
```suggestion
bool IsSinkingEnabled(ov::Node*);
```

**Review by nkogteva — CHANGES_REQUESTED** [review]
Something is wrong with onednn sunmodules, please rebase carrefully

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking_general.cpp (hunk 3 above)
Comment (evkotov) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking_general.cpp:39]: I don't see the difference )

**Review by nkogteva — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

**Pull request merged**